File: Бойко Дмитро/Лабораторна робота №1/Лабораторна робота №1/Operator.cs
using Lab_1;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab_1
{
    class Operator
    {
        public int ID { get; set; }
        public double TalkingCharge { get; set; }
        public double MessageCost { get; set; }
        public double NetworkCharge { get; set; }
        public int DiscountRate { get; set; }

        public Operator(int id, double talkingCharge, double messageCost, double networkCharge, int discountRate)
        {
            ID = id;
            TalkingCharge = talkingCharge;
            MessageCost = messageCost;
            NetworkCharge = networkCharge;
            DiscountRate = discountRate;
        }

        public double CalculateTalkingCost(int minute, Customer customer)
        {
            double cost = minute * TalkingCharge;
            if (customer.Age < 18 || customer.Age > 65)
            {
                cost *= (1 - DiscountRate / 100.0);
            }
            return cost;
        }
        public double CalculateMessageCost(int quantity, Customer customer, Customer other)
        {
            double cost = quantity * MessageCost;
            if (customer.Operators != null && other.Operators != null && customer.Operators.Length > 0)
            {
                if (customer.Operators[0] == other.Operators[0])
                {
                    cost *= (1 - customer.Operators[0].DiscountRate / 100.0);
                }
            }
            return cost;
        }
        public double CalculateNetworkCost(double amount)
        {
            return amount * NetworkCharge;
        }
    }
}

File: Пащук Нікіта/Lab1/Lab_1/Operator.py
class Operator:
    def __init__(self, id: int, talking_charge: float, message_cost: float, network_charge: float, discount_rate: int):
        self.id = id
        self.talkingCharge = talking_charge
        self.messageCost = message_cost
        self.networkCharge = network_charge
        self.discountRate = discount_rate

    def calculate_talking_cost(self, minute: int, customer):
        talking_cost = minute * self.talkingCharge
        if 65 > customer.age < 18:
            talking_cost -= talking_cost * self.discountRate
        return talking_cost

    def calculate_message_cost(self, quantity: int, customer, other_customer):
        total_message_cost = quantity * self.messageCost
        if customer.Operator == other_customer.Operator:
            total_message_cost -= total_message_cost * self.discountRate
        return total_message_cost

    def calculate_network_cost(self, amount: float):
        return amount * self.networkCharge
